pandas/core/internals/blocks.py

```diff
@@ -739,7 +739,11 @@ def to_native_types(self, slicer=None, na_rep='nan', quoting=None,
         mask = isna(values)
 
         if not self.is_object and not quoting:
-            values = values.astype(str)
+            if na_rep is not None and isinstance(na_rep, (str, bytes)):
```

Comment:
na_rep can *only* be a string; I think you are safe to only have the fixed string formatting